hack/update-openapi-spec.sh

```
@@ -0,0 +1,317 @@
+mkdir -p "${OPENAPI_ROOT_DIR}"
+curl -w "\n" -kfsS "${base_url}/openapi/v2" \
+  | jq -S '.info.version="unversioned"' \
+  > "${OPENAPI_ROOT_DIR}/swagger.json"
+
+# Fetch OpenAPI v3 discovery document
+# Note: We strip the hash query parameters from serverRelativeURL values because
+# these hashes are non-deterministic (generated at runtime by the OpenAPI aggregator)
+# and would cause spurious diffs on every run even when the actual API content is unchanged.
+echo "Updating ${DISCOVERY_ROOT_DIR}/v3-discovery.json"
+mkdir -p "${DISCOVERY_ROOT_DIR}"
+rm -rf "${DISCOVERY_ROOT_DIR}"/*
```

Shellcheck (0.11.0)

[warning] Use "${var:?}" to ensure this never expands to /* .

(SC2115)